pkgs/applications/editors/vim/plugins/vim-utils.nix

```diff
@@ -488,34 +488,37 @@ rec {
 
   toVimPlugin =
     drv:
-    drv.overrideAttrs (oldAttrs: {
-      name = "vimplugin-${oldAttrs.name}";
-      # dont move the "doc" folder since vim expects it
-      forceShare = [
-        "man"
-        "info"
-      ];
-
-      nativeBuildInputs =
-        oldAttrs.nativeBuildInputs or [ ]
-        ++ lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
-          vimGenDocHook
+    drv.overrideAttrs (
+      finalAttrs: oldAttrs: {
+        name = "vimplugin-${oldAttrs.name}";
+        # dont move the "doc" folder since vim expects it
+        forceShare = [
+          "man"
+          "info"
         ];
 
-      doCheck = oldAttrs.doCheck or true;
+        nativeBuildInputs =
+          oldAttrs.nativeBuildInputs or [ ]
+          ++ lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+            vimGenDocHook
+          ];
 
-      nativeCheckInputs =
-        oldAttrs.nativeCheckInputs or [ ]
-        ++ lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
-          vimCommandCheckHook
-          # many neovim plugins keep using buildVimPlugin
-          neovimRequireCheckHook
-        ];
+        doCheck = oldAttrs.doCheck or true;
 
-      passthru = (oldAttrs.passthru or { }) // {
-        vimPlugin = true;
-      };
-    });
+        nativeCheckInputs =
+          oldAttrs.nativeCheckInputs or [ ]
+          ++ lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) (
+            [
+              vimCommandCheckHook
+            ]
+            ++ lib.optional (finalAttrs.finalPackage ? nvimRequireCheck) neovimRequireCheckHook
```

Comment:
Is this limiting the running to explicit `nvimRequireChecks` in overrides?

Reply:
Looks like this just reverts https://github.com/NixOS/nixpkgs/pull/352277 essentially

Reply:
+1, you should check for `doCheck = false` instead of `nvimRequireCheck`

Reply:
Passing doCheck = false already skips running the hooks in native check inputs. I don't know if there's a different behavior by explicitly adding it here

Reply:
I haven't looked at #352277 so without the hook it wouldn't work ? my change made a lot of sense before realizing most of it was already available in staging. Looks like this PR is not needed after all, worse it degrades the automatic checks ?